pkg/util/cluster/cluster.go

```diff
@@ -172,6 +181,26 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		return err
 	}
 
+	keyVaultNameResourceGroup := vnetResourceGroup
+	if len(keyVaultNameResourceGroup) > 15 {
+		keyVaultNameResourceGroup = vnetResourceGroup[:14]
+	}
+
+	kvName := fmt.Sprintf("%s%s", keyVaultNameResourceGroup, sharedKeyVaultNameSuffix)
```

Comment:
`keyVaultNameResourceGroup` is a bit confusing: has nothing to do with key vault resource group. Maybe something like below?

Also happy for this to be in a smaller up PR.

```suggestion
	var kvName string
	if len(vnetResourceGroup) > 15 {
		// Potentially some comment explaining why
		kvName = vnetResourceGroup[:14] + sharedKeyVaultNameSuffix
	} else {
		kvName = vnetResourceGroup + sharedKeyVaultNameSuffix
	}
	
```